problem2/has_triplicate.pl

```diff
@@ -7,25 +7,42 @@
 % [1,2,2,2,1,1,1,3]
 % ------------------------------
 
+% ---- member(List,A) ---- 
+% member/2 evaluates to True when A is a member of List
+
 member([H|_], H).
 
 member([_|T], X) :-
   member(T,X).
 
+% ---- duplicate(List,A) ---- 
+% duplicate/2 evaluates to True when A appears at least twice in List
+
 duplicate([H|T], H) :- 
   member(T,H).
 
 duplicate([_|T], X) :-
   duplicate(T,X).
 
 % loop([H|T]) :-
-%   member(T,H), write(H) ;
-%   loop(T).
+%  member(T,H), write(H) ;
+%  loop(T).
+
+% loop([H|T]) :-
+%  member(T,H) ;
+%  loop(T).
+
+% ---- is_triplicate(List,A) ----
+% is_triplicate/2 evaluates to True if A appears at least thrice in List and subsequently prints A
 
-loop([H|T]) :-
-  member(T,H) ;
-  loop(T).
+is_triplicate([H|T],U) :- 
+  /+member(H,U),
+  duplicate(T, H),
+  write(H),
+  append(H,U);
```

Comment:
The standard append function throws an error. The log is as below:
```
ERROR:   [17] throw(error(type_error(list,a),_7340))
ERROR:   [12] lists:append(a,[]) at /usr/lib/swi-prolog/library/lists.pl:134
ERROR:   [11] is_triplicate([a,a|...],[]) at /home/shreyans/Desktop/prolog.pl:16
ERROR:    [9] <user>
```
As in the second line, `append` expects a list as the first argument, but gets a single element